pyDeltaRCM/water_tools.py

```diff
@@ -392,8 +439,124 @@ def build_weight_array(self, array, fix_edges=False, normalize=False):
         return wgt_array
 
 
+@njit
+def _get_weight_at_cell_water(ind, weight_sfc, weight_int, depth_nbrs, ct_nbrs,
+                              dry_depth, gamma, theta):
+
+    # create a fixed set of bools
+    dry = (depth_nbrs < dry_depth)
+    wall = (ct_nbrs == -2)
+    ctr = (np.arange(9) == 4)
+    drywall = np.logical_or(dry, wall)
+    invalid = np.logical_or(wall, ctr)
+    if ind[0] == 0:
+        invalid[:3] = True
+
+    # set drywall to zero
+    weight_sfc[drywall] = 0
+    weight_int[drywall] = 0
+
+    # set ctr to 0 before rebalancing
+    weight_sfc[ctr] = 0
+    weight_int[ctr] = 0
+
+    # sanity check
+    if np.any(np.isnan(weight_sfc)) or np.any(np.isnan(weight_int)):
+        raise RuntimeError('NaN encountered in input to water weighting. '
+                           'Please report error.')
+
+    # initial rebalance weights
+    if np.sum(weight_sfc) > 0:
+        weight_sfc = weight_sfc / np.sum(weight_sfc)
+    if np.sum(weight_int) > 0:
+        weight_int = weight_int / np.sum(weight_int)
+
+    weight = gamma * weight_sfc + (1 - gamma) * weight_int
+    weight = (depth_nbrs ** theta) * weight
+
+    # enforce disallowed choice to not move
+    weight[ctr] = 0
+
+    # correct the weights for random choice
+    if np.sum(weight) > 0:
+        # if any cells have positive weights, rebalance to sum() == 1
+        weight = weight / np.sum(weight)
+
+    elif np.sum(weight) == 0:
+        # if all weights are zeros/nan
```

Comment:
Yeah, I will revise the comment and add a check for nans in this set of `if-then`s. FWIW I removed the "safety" of the `np.nansum()` because 1) it made it hard to read the logic with `0`s and `np.nan`s, and 2) the `nansum` is *way* slower than the `sum`.